contracts/protocols/mainnet/erc4626/main.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import "./interfaces.sol";
+
+contract Resolver {
+    struct VaultData {
+        bool isToken;
+        string name;
+        string symbol;
+        uint256 decimals;
+        address asset;
+        uint256 totalAssets;
+    }
+
+    function getVaultDetails(address vaultAddress) public view returns (VaultData memory) {
```

Comment:
Also, add one more function as to fetch mutliple vault's detail at once.
`getVaultsDetails(address[] vaultAddresses)`